Support for class properties

Deukhoofd opened the pull request:
Adds support for the new class properties introduced in Tiled 1.8, as described [here](https://doc.mapeditor.org/en/stable/reference/tmx-map-format/#property).

I had to write some slightly hacky code to work around the fact that its child tag is optional, which would break the parser when it tried to parse it anyway.

The new property value ClassValue differs from the other PropertyValues in that its a struct enum variant, instead of a tuple one. I did this due to the additional property type that is included in it. Having it as a struct variant feels cleaner to me than having it as a two element tuple.

Deukhoofd committed:
Support for class properties

aleokdev commented:
Thanks for the PR! I'll look at this on PC whenever I have time

aleokdev commented on assets/tiled_class_property.tmx:
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<map version="1.5" tiledversion="1.7.2" orientation="orthogonal" renderorder="right-down" width="2" height="2" tilewidth="32" tileheight="32" infinite="0" nextlayerid="3" nextobjectid="4">
```
Wasn't `class` added on Tiled 1.8? How was this file made?

Deukhoofd replied:
I modified the tiled_object_property.tmx using values from a map file of my own. Generating a full file for testing through Tiled would probably have been cleaner.

bjorn replied:
I guess we can leave it as-is now, since it's not relevant for the test.

aleokdev reviewed (dismissed):
Looks fine to me, I'll let @bjorn also take a look at it

bjorn commented on src/properties.rs:
```diff
@@ -135,15 +143,31 @@ pub(crate) fn parse_properties(
+                    HashMap::new()
+                };
+                p.insert(k, PropertyValue::ClassValue {
+                    property_type: p_t.unwrap_or_else(|| "class".to_owned()),
```
I think it would be better to default to an empty string than to "class".

bjorn replied:
I think it would be better to default to an empty string than to "class".
```suggestion
                    property_type: p_t.unwrap_or_default(),
```

bjorn reviewed (commented):
Looks good to me too! I had just one suggestion.

Deukhoofd committed:
Apply suggestions from PR

- Open and save the test map so we know for certain it's in line with Tiled output
- If we can't find a property type on a class property, default to an empty string instead of "class"

aleokdev commented:
Only thing missing are the changes in the CHANGELOG.md file, after that it's good to go! Thanks!

Deukhoofd committed:
Update Changelog

aleokdev commented:
I forgot this is a breaking change (adds new variant to PropertyValue), so it should be based against `next`, not current, sorry!

bjorn commented:
@Deukhoofd I've tried to change the base branch of the PR, but it's a bit tricky, since you've created it directly from your `current` branch. Unfortunately, it is not possible to change the source branch of a PR, so either you'd need to temporary point your `current` to `next` or we need a new PR.

Or we could just use this PR to update the `next` branch with the other changes on `current`... though that's a bit messy.

aleokdev commented:
Couldn't these changes be rebased on next?
